### PR TITLE
DASHBUILDE-60: Shared JSON marshalling services

### PR DESCRIPTION
**Pull request by dgutierr**
Make the JSON marshalling services for DataSet, DataSetLookup & DataSetDef instances reusable and available as a shared services in order to not only foster reusability but also to being able to unit test dependent code both on client and server side implementations.

### CODE REVIEW

File: dashbuilder-shared/dashbuilder-displayer-api/src/test/java/org/dashbuilder/displayer/json/DisplayerSettingsJsonTest.java [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+    public void testByCountrySettings() {
+        _testSettingsMarshalling(OPPS_BY_COUNTRY);
+    }
+
+    @Test
+    public void testAllOppsSettings() {
+        _testSettingsMarshalling(OPPS_ALLOPPS_LISTING);
+    }
+
+    public void _testSettingsMarshalling(DisplayerSettings settings) {
```
Comment (jhrcek) [inline dashbuilder-shared/dashbuilder-displayer-api/src/test/java/org/dashbuilder/displayer/json/DisplayerSettingsJsonTest.java]: Seems like ideal place to use JUnit Parameterized Tests

Comment (dgutierr) [inline dashbuilder-shared/dashbuilder-displayer-api/src/test/java/org/dashbuilder/displayer/json/DisplayerSettingsJsonTest.java]: Done as suggested :-)

**Comment by dgutierr** [comment]
This PR forces to upgrade a dependency to the new "dashbuilder-json" module is some downstream repos:

https://github.com/droolsjbpm/guvnor/pull/172
https://github.com/droolsjbpm/jbpm-console-ng/pull/204
https://github.com/droolsjbpm/kie-wb-distributions/pull/125

All those PR shall be merged at the same time.

**Comment by csadilek** [comment]
+1, looks good. Also merging this right away as I've merged the downstream PRs which depend on this to go in.

**Comment by csadilek** [comment]
Thanks!

master: e23a85b93687096079383b6e444ae8418a363272

**Pull request closed without merging**